controllers/kubeadmconfig_controller.go

```diff
@@ -50,21 +50,27 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 
 	config := kubeadmv1alpha2.KubeadmConfig{}
 	if err := r.Get(ctx, req.NamespacedName, &config); err != nil {
-		log.Error(err, "failed to get config", "stacktrace", fmt.Sprintf("%+v", err))
+		log.Error(err, "failed to get config")
 		return ctrl.Result{}, err
 	}
 
+	// bail super early if it's already ready
+	if config.Status.Ready {
+		log.Info("ignoring an already ready config")
+		return ctrl.Result{}, nil
+	}
+
 	// Find the owner reference
 	var machineRef *v1.OwnerReference
 	for _, ref := range config.OwnerReferences {
-		if ref.Kind == machineKind {
+		if fmt.Sprintf("%s, Kind=%s", ref.APIVersion, ref.Kind) == machineKind {
```

Comment:
I'd almost rather see two separate comparisons than string munging to a single comparison.